### DISCUSSION TITLE
Can't make Authentik OAuth2 work with Wiki JS

### DISCUSSION THREAD

**Discussion by EHRETic** [requarks/wiki#6909] [Guides]
Hello, everything is in the title! I don't know what I'm doing wrong so I ask on the WikiJS side too.

This is a copy paste of the Authentik topic I opened here: https://github.com/goauthentik/authentik/issues/7974

**Describe the bug**
Despite I was able to configure a working Portainer Oauth, I can't get WikiJS to work with same user

**To Reproduce**
1. Configure WikiJS Oauth2 generic OpenID
2. Try to login
3. Wiki site redirects me to auth
4. Auth works bur access to WikiJS fails

**Expected behavior**

**Screenshots**
Configuration of WikiJS (wich is not a one to one match of the documentation (some extra fields
![image](https://github.com/goauthentik/authentik/assets/28634810/8b8fb29d-ff49-4590-88d1-8c19b9eb66f8)

Error
![image](https://github.com/goauthentik/authentik/assets/28634810/c730b312-d9b2-4f4b-a5d2-b7f556dbdc05)

**Logs**
<details>
    <summary>Stacktrace from authentik</summary>

```
Traceback (most recent call last):
  File "/ak-root/venv/lib/python3.11/site-packages/rest_framework/views.py", line 497, in dispatch
    self.initial(request, *args, **kwargs)
  File "/ak-root/venv/lib/python3.11/site-packages/rest_framework/views.py", line 414, in initial
    self.perform_authentication(request)
  File "/ak-root/venv/lib/python3.11/site-packages/rest_framework/views.py", line 324, in perform_authentication
    request.user
  File "/ak-root/venv/lib/python3.11/site-packages/rest_framework/request.py", line 227, in user
    self._authenticate()
  File "/ak-root/venv/lib/python3.11/site-packages/rest_framework/request.py", line 380, in _authenticate
    user_auth_tuple = authenticator.authenticate(self)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/ak-root/venv/lib/python3.11/site-packages/rest_framework/authentication.py", line 130, in authenticate
    self.enforce_csrf(request)
  File "/ak-root/venv/lib/python3.11/site-packages/rest_framework/authentication.py", line 148, in enforce_csrf
    raise exceptions.PermissionDenied('CSRF Failed: %s' % reason)
rest_framework.exceptions.PermissionDenied: CSRF Failed: Origin checking failed - https://auth-test.mydomain.ext does not match any trusted origins.
```
</details>

**Version and Deployment (please complete the following information):**
- authentik version: 2023.10.5
- Docker container with macvlan IP adresses

**Additional context**
- I use LetsEncrypt certificates for both instances (authentik & wikijs), both are behind an NGINX reverse proxy
- A Portainer instance works fine
- I use Incognito mode for testing
- all my users are AD based (but I don't think it matters)

Maybe some of you did already configure this successfully, any clue where I should start looking? Thanks in advance 😉

**Comment by andymarden** [comment]
I have it working. I use the self-signed cert and it seems to work fine without me having to install anything on the wiki.js side.

Other things I can see that are different to what you have:

- I use the default bridge interface for the wiki.js container
- I have not got "Skip User Profile" on
- DisplayName is just name

That seems to be everything on the wikijs side.

Did you remember to grant the user or group they belong to as having access to the wikijs application within Authentik?

**Comment by EHRETic** [comment]
Hi,

In fact, the solution was to add the following line in NGINX configuration (both entities, Authentik & Wiki):

`proxy_set_header X-Forwarded-Proto $scheme;`

I had nothing to change on WikiJS side.
